OpenRA.Mods.Common/Traits/Attack/AttackFollow.cs

```diff
@@ -121,10 +123,20 @@ public override Activity Tick(Actor self)
 				}
 
 				if (!onRailsHack)
-					attack.Target = Target.Invalid;
+					attack.Key.Target = Target.Invalid;
 
 				return NextActivity;
 			}
+
+			IEnumerable<KeyValuePair<AttackFollow, Armament>> SelectActiveAttack(Target t, bool forceAttack)
```

Comment:
Do you plan on using this method in any other places than the current one (and still requiring a whole enumerable)?
Otherwise you could just return the first fitting pair and bail out early.